### PR TITLE
Issue #13: Support relative source reference

### PR DESCRIPTION
**Pull request by MedMaalej**
- Add ReferenceResolverProcessor
- Update ConnectorLibraryParser
- Create unit tests

### CODE REVIEW

File: metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/NodeProcessorHelper.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public static AbstractNodeProcessor withExtendsAndConstantsProcessor(
 			.builder()
 			.connectorDirectory(connectorDirectory)
 			.mapper(mapper)
-			.next(constantsProcessor())
+			.next(ReferenceResolverProcessor.builder().next(constantsProcessor()).build())
```
Comment (NassimBtk) [inline metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/NodeProcessorHelper.java:60]: Please simplify:
`.next(new ReferenceResolverProcessor(constantsProcessor()))`

File: metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/ReferenceResolverProcessor.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	@Override
+	public JsonNode process(JsonNode node) throws IOException {
+		return super.process(node);
+	}
+
+	/**
+	 * Processes a given Json node by calling {@link JsonNodeUpdater}
+	 * @param node The JsonNode to be processed.
+	 * @return {@link JsonNode} instance
+	 * @throws IOException
```
Comment (NassimBtk) [inline metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/ReferenceResolverProcessor.java:70]: Complete javadoc.

File: metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/TemplateVariableProcessor.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public JsonNode processNode(JsonNode node) throws IOException {
 
 		// Call JsonNodeUpdater to replace the placeholder by the variable value
 		JsonNodeUpdater
-			.builder()
+			.jsonNodeUpdaterBuilder()
```
Comment (NassimBtk) [inline metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/TemplateVariableProcessor.java:66]: Should be `builder()`

File: metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/AbstractJsonUpdater.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		this.predicate = predicate;
+	}
+
+	@NonNull
+	protected final JsonNode jsonNode;
+
+	@NonNull
+	protected final Predicate<String> predicate;
+
+	protected abstract void update();
```
Comment (NassimBtk) [inline metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/AbstractJsonUpdater.java:43]: Add javadc.

**Review by NassimBtk — APPROVED** [review]

**Review by alexdgilbert — APPROVED** [review]

**Pull request merged**